### ISSUE TITLE
very slow response [404 Not Found] s3.HeadObject only on one pod after some time

### ISSUE THREAD

**Issue by WMP** [minio/minio#20357]
## NOTE

I have minio in 4 replicas on k8s. I don't know why, but there is always one under which I receive queries which returns the message that the file was not found only after 59-60s. A restart of the pod helps for 10 minutes, then the problem occurs again. Here is the trace log with grep on '404  Not':

```
2024-08-31T08:30:24.126 [404 Not Found] s3.HeadObject minio-3.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222554.pdf MyIP    1.88ms       ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:29:24.125 [404 Not Found] s3.HeadObject minio-0.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222554.pdf MyIP    1m0.005025s  ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:30:25.031 [404 Not Found] s3.HeadObject minio-1.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222555.pdf MyIP    2.391ms      ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:30:25.143 [404 Not Found] s3.HeadObject minio-2.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222559.pdf MyIP    2.68ms       ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:31:18.716 [404 Not Found] storageR.ReadXL minio-0.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=270f53c9-4a71-4828-9c29-71a50e048362&file-path=rebalance.bin&read-data=true&volume=.minio.sys  10.42.1.177      136µs       ⇣  134.809µs  ↑ 58 B ↓ 14 B
2024-08-31T08:31:18.733 [404 Not Found] storageR.ReadXL minio-0.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=270f53c9-4a71-4828-9c29-71a50e048362&file-path=config%2Ftier-config.bin&read-data=true&volume=.minio.sys  10.42.1.177      58µs        ⇣  57.57µs   ↑ 58 B ↓ 14 B
2024-08-31T08:31:18.859 [404 Not Found] storageR.ReadXL minio-0.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=270f53c9-4a71-4828-9c29-71a50e048362&file-path=config%2Fsite-replication%2Fstate.json&read-data=true&volume=.minio.sys  10.42.1.177      127µs       ⇣  124.489µs  ↑ 58 B ↓ 14 B
2024-08-31T08:30:25.188 [404 Not Found] s3.HeadObject minio-0.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222560.pdf MyIP    1m0.041922s  ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:31:25.481 [404 Not Found] s3.HeadObject minio-3.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222560.pdf MyIP    48.755ms     ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:31:25.588 [404 Not Found] s3.HeadObject minio-1.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222562.pdf MyIP    2.124ms      ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:31:25.691 [404 Not Found] s3.HeadObject minio-2.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222566.pdf MyIP    2.535ms      ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:32:02.745 [404 Not Found] storageR.ReadXL minio-0.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=270f53c9-4a71-4828-9c29-71a50e048362&file-path=config%2Ftier-config.bin&read-data=true&volume=.minio.sys  10.42.2.225      128µs       ⇣  126.708µs  ↑ 58 B ↓ 14 B
2024-08-31T08:32:02.746 [404 Not Found] storageR.ReadXL minio-1.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=b00c34a8-17eb-47d7-886b-a847a4574311&file-path=config%2Ftier-config.bin&read-data=true&volume=.minio.sys  10.42.2.225      151µs       ⇣  149.207µs  ↑ 58 B ↓ 14 B
2024-08-31T08:32:02.746 [404 Not Found] storageR.ReadXL minio-2.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=1619954b-b0f4-45c2-b2d7-5f3820ec6e21&file-path=config%2Ftier-config.bin&read-data=true&volume=.minio.sys  10.42.2.225      121µs       ⇣  118.401µs  ↑ 58 B ↓ 14 B
2024-08-31T08:32:25.806 [404 Not Found] s3.HeadObject minio-3.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222569.pdf MyIP    2.157ms      ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:31:25.830 [404 Not Found] s3.HeadObject minio-0.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222569.pdf MyIP    59.999833s   ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:32:25.840 [404 Not Found] s3.HeadObject minio-1.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222570.pdf MyIP    2.696ms      ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:32:25.935 [404 Not Found] s3.HeadObject minio-2.minio-headless.minio-namespace.svc.cluster.local:9000/minio-namespace/api/public/00222573.pdf MyIP    2.964ms      ⇣  0s         ↑ 250 B ↓ 0 B
2024-08-31T08:32:39.030 [404 Not Found] storageR.ReadXL minio-3.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=65b15198-8cba-4a94-afd9-acbb554a6871&file-path=config%2Ftier-config.bin&read-data=true&volume=.minio.sys  10.42.2.196      130µs       ⇣  128.669µs  ↑ 58 B ↓ 14 B
2024-08-31T08:32:39.031 [404 Not Found] storageR.ReadXL minio-1.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=b00c34a8-17eb-47d7-886b-a847a4574311&file-path=config%2Ftier-config.bin&read-data=true&volume=.minio.sys  10.42.2.196      176µs       ⇣  174.208µs  ↑ 58 B ↓ 14 B
2024-08-31T08:32:39.031 [404 Not Found] storageR.ReadXL minio-2.minio-headless.minio-namespace.svc.cluster.local:9000/minio/storage/bitnami/minio/data/v57/readxl?disk-id=1619954b-b0f4-45c2-b2d7-5f3820ec6e21&file-path=config%2Ftier-config.bin&read-data=true&volume=.minio.sys  10.42.2.196      105µs       ⇣  104.726µs  ↑ 58 B ↓ 14 B
```

The entire log is 8MB, what other lines near these 404s might you be interested in?

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used (`minio --version`): 14.6.2
* Server setup and configuration: Qemu VM on bare metal
* Operating System and version (`uname -a`): Ubuntu 22.04

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by klauspost** [comment]
`14.6.2` is not a minio version. Upgrade to latest.

**Comment by WMP** [comment]
Sorry, I specified the version of chart.... I am using this docker: docker.io/bitnami/minio:2024.5.28-debian-12-r0

**Comment by klauspost** [comment]
Upgrade. 60s is the timeout value of these requests - your network connections are hanging. We have made several tweaks to that since your version.
